pkg/virt-controller/watch/export/export.go

```diff
@@ -403,17 +402,21 @@ func (ctrl *VMExportController) handleIsSourcePvc(vmExport *exportv1.VirtualMach
 }
 
 func (ctrl *VMExportController) handleIsVmSnapshot(vmExport *exportv1.VirtualMachineExport, service *corev1.Service) (time.Duration, error) {
-	var retry time.Duration
-	pvcs := make([]*corev1.PersistentVolumeClaim, 0)
+	var pvcs []*corev1.PersistentVolumeClaim
 
 	vmSnapshot, exists, err := ctrl.getVmSnapshot(vmExport.Namespace, vmExport.Spec.Source.Name)
 	restoreableSnapshots := 0
 	if err != nil {
 		return 0, err
 	} else if exists {
-		pvcs, restoreableSnapshots, err = ctrl.handlePVCsForVirtualMachineSnapshot(vmExport, vmSnapshot)
-		if err != nil {
-			return 0, err
+		if vmSnapshot.Status.ReadyToUse != nil && *vmSnapshot.Status.ReadyToUse {
+			pvcs, restoreableSnapshots, err = ctrl.handlePVCsForVirtualMachineSnapshot(vmExport, vmSnapshot)
+			if err != nil {
+				return 0, err
+			}
+		} else {
+			// snapshot is not ready yet, try again in a second.
+			return time.Second, nil
```

Comment:
one second seems like a little much.  how about 2 or 3 secs?  Or watch VMsnapshots and add a "vmsnapshot" index to VMexport informer?